### ISSUE TITLE
Private fields in java class exposed in python

### ISSUE THREAD

**Issue by rusu24edward** [jpype-project/jpype#406]
## Context
jpype seems to expose the private fields of my java class when I create a jpype.JClass for it to use with Python. I believe this issue may be related to an old one from 2005 (https://sourceforge.net/p/jpype/bugs/39/). In that ticket, Steve Menard mentioned that the private variables coming through was a bug and that he would open a ticket; I couldn't find that ticket.

Here's my simple case

``` java
// Java class
package testpackage;
public class TestClassWrapper {
  public TestClassWrapper(double value) { // public constructor I want the user to have
    internal = new Double(value);
  }
  public void MultiplyDouble(double multiplyBy) { // public function I want the user to have
    internal *= multiplyBy;
  }
  double getDouble() { // package encapsulated
    return -10.0;
  }
  private Double internal; // internal field I do not want the user to see
}
```
Then in Python I have
``` python
# python file
# Launch the jvm and setup jpype
TestClass = jpype.JClass("testpackage.TestClassWrapper")
myTestClass1 = TestClass(2.0)
print myTestClass1.internal
myTestClass1.MultiplyBy(3.0)
print myTestClass1.internal    
print myTestClass1.getDouble()
```

When I run this python file, here's the output
``` python
2.0 # print out from myTestClass1.internal
6.0 # print out from myTestClass1.internal
Error: Error in Python input # Attempt to access getDouble()
Traceback (most recent call last): print myTestClass1.getDouble()
raise ex AttributeError: 'pythoninterface.model.TestClassWrapper' object has no attribute 'getDouble'
```

Basically, I just think it's strange that python cannot access a package-encapsulated function (rightfully so), yet it can access private members. This seems like a bug to me.

## What I've done
I dug around in the jpype code a bit to see what I could find. I started at _jclass.py -> JClass since that's the function for access to the TestClassWrapper. This leads me down to _JavaClass->__new__. There's a nice little comment in there that says "#add the field". To confirm, I put a print statement for fname, and it indeed is grabbing the private fields.

So then I looked around for getClassFields since that seems to be the function doing all the work. I found PyJPClass::getClassFields in ./native/python/py_class.cpp. Looks like it grabs all static and all instance fields. I wonder if there should be a check in here to only output public fields?

## Work Around
For the time being, I'm just using a hack in _JavaClass->__new__ that checks for the field name, and if it is "internal", don't add it to the members list.

**Comment by Thrameos** [comment]
I believe this was addressed in the devel branch.   It was a bug in the C++ layer that failed to check if the declared members were private or public.

--Karl

**Comment by rusu24edward** [comment]
@Thrameos thanks for the quick response. Any thoughts on when the next release will be?

**Comment by Thrameos** [comment]
It should have been 5 months ago, but I have been out of commission due to issues at my work place and  home.  I am trying to get back in by May, but if anyone would like to try the devel version and submit issues or patches it would certainly help.

**Comment by rusu24edward** [comment]
@Thrameos thanks for your hard work on this! I will try to compile and run with the devel branch if I get a chance.

**Comment by Thrameos** [comment]
Fixed in 0.7 closing for now.
